#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE_NAME = "squirrel_navigation"

gen = ParameterGenerator()

# PID gains
gen.add("verbose", bool_t, 0, "Set verbose", False)
gen.add("max_linear_vel", double_t, 0, "Maximum linear speed allowed to the robot", 0.3, 0.0, 0.7)
gen.add("max_angular_vel", double_t, 0, "Maximum angular speed allowed to the robot", 0.5, 0.0, 1.0)
gen.add("xy_goal_tolerance", double_t, 0, "Goal reaching tolerance (linear distance)", 0.05, 0.01, 0.25)
gen.add("yaw_goal_tolerance", double_t, 0, "Goal reaching tolerance (angular distance)", 0.05, 0.01, 0.75)
gen.add("odom_topic", str_t, 0, "odometry topic", "/odom")

exit(gen.generate(PACKAGE_NAME, "squirrel_navigation", "LocalPlannerPlugin"))